obspy/io/nied/knet/core.py

```diff
@@ -0,0 +1,312 @@
+    # the trace as miniSEED; if convert_stnm is enabled, the last two
+    # letters of the station code are written to the 'location' field
+    stnm = flds[2]
+    location = ''
+    if convert_stnm and len(stnm) > 5:
+        location = stnm[-2:]
+        stnm = stnm[:-2]
+    if len(stnm) > 7:
+        raise KnetFormatError("Station name can't be more than 7 characters\
+        long!")
+    hdrdict['station'] = stnm
+    hdrdict['location'] = location
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    hdrdict['knet']['stla'] = float(flds[2])
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    hdrdict['knet']['stlo'] = float(flds[2])
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    hdrdict['knet']['stel'] = float(flds[2])
+
+    # Data information
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    dt = flds[2] + ' ' + flds[3]
+    # A 15 s delay is added to the record time by the
+    # the K-NET and KiK-Net data logger
+    dt = UTCDateTime(datetime.strptime(dt, '%Y/%m/%d %H:%M:%S')) - 15.0
+    # All times are in Japanese standard time which is 9 hours ahead of GMT
+    dt -= 9 * 3600.
+    hdrdict['starttime'] = dt
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    freqstr = flds[2]
+    m = re.search('[0-9]*', freqstr)
+    freq = int(m.group())
+    delta = 1.0 / freq
+    hdrdict['delta'] = delta
+    hdrdict['sampling_rate'] = freq
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    hdrdict['knet']['duration'] = float(flds[2])
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    channel = flds[1].replace('-', '')
+    kiknetcomps = {'1': 'NS1', '2': 'EW1', '3': 'UD1',
+                   '4': 'NS2', '5': 'EW2', '6': 'UD2'}
+    if channel.strip() in kiknetcomps.keys():  # kiknet directions are 1-6
+        channel = kiknetcomps[channel.strip()]
+    hdrdict['channel'] = channel
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    eqn = flds[2]
+    num, denom = eqn.split('/')
+    num = float(re.search('[0-9]*', num).group())
+    denom = float(denom)
+    # convert the calibration from gal to m/s^2
+    hdrdict['calib'] = 0.01 * num / denom
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    acc = float(flds[3])
+    hdrdict['knet']['accmax'] = acc
+
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    dt = flds[2] + ' ' + flds[3]
+    dt = UTCDateTime(datetime.strptime(dt, '%Y/%m/%d %H:%M:%S'))
+    # All times are in Japanese standard time which is 9 hours ahead of GMT
+    dt -= 9 * 3600.
+    hdrdict['knet']['last correction'] = dt
+
+    # The comment ('Memo') field is optional
+    _i += 1
+    flds = _prep_hdr_line(hdrnames[_i], hdrlines[_i])
+    if len(flds) > 1:
+        hdrdict['knet']['comment'] = ' '.join(flds[1:])
+
+    if len(hdrlines) != _i + 1:
```

Comment:
Shouldn't this be checked at the beginning of the function? What if the number of header numbers is too short?